swan-lake/learn-the-platform/test-document-the-code/debug-ballerina-programs.md

```diff
@@ -7,262 +7,22 @@ permalink: /learn/debug-ballerina-programs/
 active: debug-ballerina-programs
 ---
 
-When writing large-scale applications with complex logic, it is quite possible to have syntax, semantic, 
-or runtime issues in your code. The Ballerina compiler comes in handy when detecting syntax and semantic issues. 
-However, it is very difficult for a compiler to detect runtime errors like logical errors because they occur during the 
-program execution after a successful compilation. This is where the dedicated debugging tooling support becomes important.
+The Ballerina compiler comes in handy to detect syntax and semantic issues that occur in your code when developing large-scale applications with complex logic. However, it is difficult for the compiler to detect runtime errors like logical errors because they occur during the program execution after a successful compilation. This is where the dedicated debugging tooling support of Ballerina becomes important.
 
-Currently, we provide a feature-rich debugging experience for troubleshooting Ballerina applications
-via the <a href="https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina" target="_blank">Ballerina Visual Studio Code extension</a>.
+## Debugging sessions
 
-## Start a debug session
+The VS Code extension provides three types of sessions to debug your Ballerina code.
 
-The Ballerina extension provides multiple options to debug Ballerina applications, and the most convenient way will be using the context-aware debug codelens support.
+>**Info:** For more information on the debug sessions, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/#debugging-sessions" target="_blank">Debug sessions</a>.
 
-However, if you require launching debug sessions with additional configurations (i.e., program arguments, environment variables), you can use `launch.json` configurations
-(For more information, see [Start a debug session with configurations](#start-a-debug-session-with-configurations)).
+## Debugging methods
 
-### Start a program debug session
+The VS Code extension provides three types of methods to execute the above debug sessions.
 
-Follow the steps below to start a quick debug session using CodeLens.
+>**Info:** For more information on the debug methods, go to <a href="https://wso2.com/ballerina/vscode/docs/debug-the-code/debugging-overview/#debugging-methods" target="_blank">Debug methods</a>.
 
-1. Open the folder which includes the Ballerina program you want to debug, and open the source file in the editor.
```

Comment:
IMO we don't need to have a separate section on debugging methods, as that is already covered under the previous section. Should we talk about the `debug configurations` instead?

Reply:
Updated.